text/0000-box-and-in-for-stdlib.md

```diff
@@ -0,0 +1,520 @@
+  invoke the `Rc::new` function. This will be more efficient for
+  building instances of `Rc<T>` when `T` is a large type.  (It is also
+  arguably much cleaner syntax to read, regardless of the type `T`.)
+
+  Note that this change will require end-user code to no longer assume
+  that `box <expr>` always produces a `Box<T>`; such code will need to
+  either add a type annotation e.g. saying `Box<_>`, or will need to
+  call `Box::new(<expr>)` instead of using `box <expr>`.
+
+* Add support for parsing `in (<place-expr>) <expr>` as the basis for the
+  placement operator.
+
+  Remove support for `box (<place-expr>) <expr>` from the parser.
+
+  Make `in (<place-expr>) <expr>` an overloaded operator that uses
+  the `<place-expr>` to determine what placement code to run.
+
+* The only stablized implementation for the `box <expr>` operator
+  proposed by this RFC is `Box<T>`. The question of which other types
+  should support integration with `box <expr>` is a library design
+  issue and needs to go through the conventions and library
+  stabilization process.
+
+  Similarly, this RFC does not propose *any* stablized implementation
+  for the `in (<place-expr>) <expr>` operator. (An obvious candidate for
+  `in (<place-expr>) <expr>` integration would be a `Vec::emplace_back`
+  method; but again, the choice of which such methods to add is a
+  library design issue, beyond the scope of this RFC.)
+
+  (A sample implementation illustrating how to support the operators
+  on other types is given in [Appendix A].)
+
+# Drawbacks
+
+* End-users might be annoyed that they cannot add implementations of
+  the overloaded-`box` and placement-`in` operators themselves. But
+  such users who want to do such a thing will probably be using the
+  nightly release channel, which will not have the same stability
+  restrictions.
+
+# Alternatives
+
+* We could keep the `box (<place-expr>) <expr>` syntax. It is hard
+  to see what the advantage of that is, unless (1.) we can identify
+  many cases of types that benefit from supporting both
+  overloaded-`box` and placement-`in`, or unless (2.) we anticipate
+  some integration with `box` pattern syntax that would motivate using
+  the `box` keyword for placement.
+
+* Do nothing. I.e. do not even accept an unstable libstd-only protocol
+  for placement-`in` and overloaded-`box`. This would be okay, but
+  unfortunate, since in the past some users have identified
+  intermediate copies to be a source of inefficiency, and proper use
+  of `box <expr>` and placement-`in` can help remove intermediate
+  copies.
+
+# Unresolved questions
+
+None
+
+# Appendices
+
+## Appendix A: sample operator traits
+[Appendix A]: #appendix-a-sample-operator-traits
+
+The goal is to show that code like the following can be made to work
+in Rust today via appropriate desugarings and trait definitions.
+
+```rust
+fn main() {
+    use std::rc::Rc;
+
+    let mut v = vec![1,2];
+    in (v.emplace_back()) 3; // has return type `()`
```

Comment:
I prefer `in(v.back()) 3`, fwiw, as "emplace" is just a longer way of conveying the same thing `in` does.
Also it could return `&mut T`, but I may be mistaken.

This is straying a bit from the topic at hand, and I'm not sure what will come of it, but I have the start of a design that makes `*v.back() = 3` possible.